storm-core/src/jvm/org/apache/storm/utils/ConfigUtils.java

```diff
@@ -0,0 +1,678 @@
+    }
+
+    public static String masterInbox(Map conf) throws IOException {
+        String ret = masterLocalDir(conf) + FILE_SEPARATOR + "inbox";
+        FileUtils.forceMkdir(new File(ret));
+        return ret;
+    }
+
+    public static String masterInimbusDir(Map conf) throws IOException {
+        return (masterLocalDir(conf) + FILE_SEPARATOR + "inimbus");
+    }
+
+    //For testing only
+    // for java
+    // try (SetMockedSupervisorLocalDir mocked = new SetMockedSupervisorLocalDir(dir)) {
+    //    run test ...
+    // }
+    //
+    // for clojure
+    // (with-open [mock (SetMockedSupervisorLocalDir. dir)]
+    //     run test ...)
+    public static class SetMockedSupervisorLocalDir implements Closeable {
+        public SetMockedSupervisorLocalDir(String dir) {
+            mockedSupervisorLocalDir = dir;
+        }
+        @Override
+        public void close() {
+            mockedSupervisorLocalDir = null;
+        }
+    }
+    private static String mockedSupervisorLocalDir = null;
+    public static String supervisorLocalDir(Map conf) throws IOException {
+        if (mockedSupervisorLocalDir != null) {
+            return null;
+        }
+        String ret = absoluteStormLocalDir(conf) + FILE_SEPARATOR + "supervisor";
+        FileUtils.forceMkdir(new File(ret));
+        return ret;
+    }
+
+    public static String supervisorIsupervisorDir(Map conf) throws IOException {
+        return ((supervisorLocalDir(conf) + FILE_SEPARATOR + "isupervisor"));
+    }
+
+    //For testing only
+    // for java
+    // try (SetMockedSupervisorStormDistRoot mocked = new SetMockedSupervisorStormDistRoot(conf)) {
+    //    run test ...
+    // }
+    //
+    // for clojure
+    // (with-open [mock (SetMockedSupervisorStormDistRoot. conf)]
+    //     run test ...)
+    public static class SetMockedSupervisorStormDistRoot implements Closeable {
+        public SetMockedSupervisorStormDistRoot(Map conf) {
+            mockedSupervisorStormDistRoot = conf;
+        }
+        @Override
+        public void close() {
+            mockedSupervisorStormDistRoot = null;
+        }
+    }
+    private static Map mockedSupervisorStormDistRoot = null;
+    public static String supervisorStormDistRoot(Map conf) throws IOException {
+        if (mockedSupervisorStormDistRoot != null) {
+            return null;
+        }
+        return stormDistPath(supervisorLocalDir(conf));
+    }
+
+    public static String supervisorStormDistRoot(Map conf, String stormId) throws IOException {
+        if (mockedSupervisorStormDistRoot != null) {
+            return null;
+        }
+        return supervisorStormDistRoot(conf) + FILE_SEPARATOR + URLEncoder.encode(stormId, "UTF-8");
+    }
+
+    public static String supervisorStormJarPath(String stormRoot) {
+        String ret = "";
+        // we do this since to concat a null String will actually concat a "null", which is not the expected: ""
```

Comment:
This pattern is repeated many, many times in this file. Can we move this to its own method and call that from here?
```
public static String supervisorStormBasePath(String stormRoot) {
        String ret = "";
        if (stormRoot != null) {
            ret = stormRoot;
        }
        return ret + FILE_SEPARATOR;
    }

public static String supervisorStormJarPath(String stormRoot) {
    return supervisorStormBasePath(stormRoot) + "stormjar.jar";
}
...
```